### ISSUE TITLE
Package structure discussion

### ISSUE THREAD

**Issue by eduardo-rodrigues** [scikit-hep/scikit-hep-tutorials#2]
It makes sense to have a bit of structure in the repository so that anyone coming here can easily find the tutorial for their topic of interest. We could largely follow the list of topics used on our website ([scikit-hep.org](http://scikit-hep.org/)) to categorise the packages: interoperability, particles and decays, histogramming, fitting, etc.

Opinions @henryiii, @HDembinski, @apuignav, @mayou36?

**Comment by HDembinski** [comment]
Sounds very good, but I wonder how to organise this. iminuit already has extensive tutorials available as part of the iminuit documentation. Should these be duplicated or moved into this repo?
Having the tutorials locally in the iminuit repo is nice, because CI can check that the tutorials don't break on a new PR. This is a very useful feature, and I am not sure how to maintain this when the tutorials are part of this repo here. Unit-testing code in the docs is very important, because they tend to break unnoticed.

**Comment by jonas-eschle** [comment]
I agree also to have it topically structured, but I think it may can also contain even larger tutorials, e.g. show how to combine particles and decays with histograms and fitting, saying real use cases.

on @HDembinski my idea:
- if you want to accomplish something and see _how skicit-hep_ can be used for that (eg generate particles and make a histogram out of it), that's what this tutorials are for
- if you want to use iminuit and know what it can do, those tutorials belong to iminuit.

There is an overlap and we could move some tutorials around, but if in doubt, we rather leave the status quo and build an additional layer here, linking maybe to other tutorials. I think though that CI is not a problem, a package could simply git clone this repo and pick the relevant tutorials for testing.

In general, I see it as a collection of "how to do (parts of) an analysis with the help of scikit-hep" examples

**Comment by eduardo-rodrigues** [comment]
Thanks everyone for the feedback. I agree very much with the comments. Just to make it super clear:

* Package-level tutorials and demos should remain with the packages, so no changes suggested there.
* Tutorials and demos showing how the different packages work best together are the ideal candidates to be put into this new repository. They are higher-level, or global, in some sense.
* By topics I did not mean package topics per se. That is why I have particles and decays as one example topic. A tutorial here would be on how to work with `DecayLanguage` and `Particle` together, for example. But sure, how to "combine particles and decays with histograms and fitting" would be a great and even larger-scope tutorial.

In short, it's more about the ecosystem and less about the individual packages.

I have the impression that we all agree. The work remains to be done, now … :-).

**Comment by eduardo-rodrigues** [comment]
To move forward: @mayou36, are you working on a demo of zfit and Scikit-HEP stuff, which we could add here?

**Comment by jonas-eschle** [comment]
Not yet, just been busy and will be unavailable for two weeks now, but afterwards this is definitely on my list and we can come back to it. I think mostly phasespace and other packages, hepunits and Particle could go well together, not sure yet how to best involve a fitting library exactly (except of iminuit of course! and the future statutils).
